### PR TITLE
Brie/updates sunset

### PR DESCRIPTION
**Pull request by briemcnally**
What does this PR do?
- [X]  Updates the latest Firebase SDK 
- [X] Transforms event names to `^[a-zA-Z0-9_]+$`
- [X] Transforms property names to `^[a-zA-Z0-9_]+$`
- [X] Fix crash when Firebase SDK is instantiated multiple times by not configuring the FIRApp if already done
- [X] Adds unit tests for new functionality
- [X] Adds working CircleCI config 

Associated JIRA Tickets: 
https://segment.atlassian.net/browse/DEST-1085

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+      # - run:
+      #     name: Fetch Cocoapods specs
+      #     command: curl https://cocoapods-specs.circleci.com/fetch-cocoapods-repo-from-s3.sh | bash -s cf
+
+      - run: make install
+      - run: make build
+      - run: make test
+      - run: make lint
+
+ # - store_test_results:
```
Comment (brennan) [inline .circleci/config.yml]: I don't know a lot of about how we test iOS in Circle - but should this job be commented out?

File: Example/Segment-Firebase.xcworkspace/xcshareddata/IDEWorkspaceChecks.plist [hunk 2]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (brennan) [inline Example/Segment-Firebase.xcworkspace/xcshareddata/IDEWorkspaceChecks.plist]: Should this file be included? It's looks like it's a warning specific to your local IDE configuration, but not sure.

File: Segment-Firebase/Classes/SEGFirebaseIntegration.m [hunk 3]
```diff
@@ -87,11 +92,26 @@ - (NSString *)formatFirebaseEventNames:(NSString *)event
                                              kFIREventSearch, @"Products Searched", nil];
 
     NSString *mappedEvent = [mapper objectForKey:event];
-
+    NSArray *periodSeparatedEvent = [event componentsSeparatedByString:@"."];
+    NSString *regexString = @"^[a-zA-Z0-9_]+$";
```
Comment (brennan) [inline Segment-Firebase/Classes/SEGFirebaseIntegration.m]: Not sure if leading/trailing spaces are handled elsewhere - this regex won't match an event with a leading or trailing space. Looks as if on Android we trim leading/trailing spaces before replacing periods or captured spaces.

Comment (briemcnally) [reply Segment-Firebase/Classes/SEGFirebaseIntegration.m]: @brennan added support to handle trimming leading and trailing whitespace to match functionality with android.

File: .circleci/config.yml (hunk 1 above)
Comment (briemcnally) [reply .circleci/config.yml]: @brennan We don't store results for any other iOS integrations so thats why I have it commented out. I will leave it that way for now.

File: Segment-Firebase/Classes/SEGFirebaseIntegration.m [hunk 4]
```diff
@@ -18,6 +18,11 @@ - (id)initWithSettings:(NSDictionary *)settings
             SEGLog(@"[FIROptions defaultOptions].deepLinkURLScheme = %@;", deepLinkURLScheme);
         }
 
+        if ([FIRApp defaultApp]) {
+            SEGLog(@"[FIRApp Configure] already called, skipping");
```
Comment (benevbright) [inline Segment-Firebase/Classes/SEGFirebaseIntegration.m]: Hey guys. Is this Ok? 
Segment-Firebase integration will still work when you skip this?

Comment (benevbright) [reply Segment-Firebase/Classes/SEGFirebaseIntegration.m]: @briemcnally  @niallzato

Comment (briemcnally) [reply Segment-Firebase/Classes/SEGFirebaseIntegration.m]: I have end to end tested these changes without any reported issues issues. 
You can reference these two PRs for more information: 
* https://github.com/segment-integrations/analytics-ios-integration-firebase/pull/37
* https://github.com/segment-integrations/analytics-ios-integration-firebase/pull/42

**Pull request merged**